README.md

```diff
@@ -1 +1,181 @@
-# Budget-app
\ No newline at end of file
+
+# 📗 Table of Contents
+
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+- [Demo](#💻-deployment)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 [Budget App] <a name="about-project"></a>
+
+**[Budgt App]** Budget App is a fully functioning website that allows a user to fully manage his/her expenses by keeping track of all payments and their related categories. For each category the user can see how much is already spent. A user can add categories and payments related to one or more categories.
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+  <ul>
+    <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
+    <li><a href="https://www.ruby-lang.org/en/">Rails</a></li>
+    <li><a href="https://www.ruby-lang.org/en/">Postgresql</a></li>
+  </ul>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **[A login page]**
+- **[Adding foods and creating records]**
+- **[A categories list view]**
+- **[Use validations for models]**
+- **[Write unit tests for models]**
+- **[Write request tests for models]**
+- **[Set up the view model]**
+- **[Use preprocessed HTML file with embedded Ruby code.]**
+- **[Use layouts and templates for shared content.]**
+
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need the following installed in your local system
+
+<ul>
+<li>Ruby</li>
+<li>Git CLI</li>
+<li>Github account</li>
+</ul>
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+  cd my-folder
+  git clone git@github.com:tajulafreen/Budget-app.git
+```
+
+### Install
+
+Install gems with:
+
+```sh
+  bundle install
+```
+
+Setup database with:
+
+```sh
+rails db:create
+rails db:migrate
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  rails server
+```
+Visit http://localhost:3000/ in your browser.
+
+### Run tests
+
+-  Install npm with:
+-  Install rspec with:
+```sh
+  bundle install
+```
+and
+
+```sh
+  rails generate rspec:install
+```
+
+run the test with:
+```sh
+  rspec spec
+```
+
+## 💻 Deployment <a name="getting-started"></a>
+- Presentation: [Presentation](https://drive.google.com/file/d/1k87d1bscWmisrPjNuKL0v9teDV7g644B/view?usp=sharing)
```

Comment:
- [ ]  Kindly please highlight a piece of code that you find interseting or something you are proud in the video presentation as stated in the requirement.